src/widget/wtracktableview.cpp

```diff
@@ -217,6 +218,11 @@ void WTrackTableView::loadTrackModel(QAbstractItemModel *model) {
         // a select() if the table is dirty.
         doSortByColumn(horizontalHeader()->sortIndicatorSection());
         return;
+    }else{
+        newModelAddr = reinterpret_cast<unsigned long>(trackModel);
```

Comment:
This is extremely dangerous code. And it is not portable. You don't even check if a unsigned long has the same size as a pointer.